src/osd/ECCommon.cc

```diff
@@ -419,14 +419,20 @@ void ECCommon::ReadPipeline::do_read_op(ReadOp &op)
     }
   }
 
+  std::optional<ECSubRead> local_read_op;
   std::vector<std::pair<int, Message*>> m;
   m.reserve(messages.size());
   for (map<pg_shard_t, ECSubRead>::iterator i = messages.begin();
        i != messages.end();
-       ++i) {
+       /* see below */) {
     op.in_progress.insert(i->first);
     shard_to_read_map[i->first].insert(op.tid);
     i->second.tid = tid;
+    if (i->first == get_parent()->whoami_shard()) {
+      local_read_op = std::move(i->second);
+      i = messages.erase(i);
```

Comment:
Nit: It isn't really necessary to erase the message which would allow you to keep the ++i in the for loop. If you are going to std::move(i->second) here then it should probably also be moved below when initializing the MOSDECSubOpRead with msg->op = i->second;

Reply:
Addressed! No other changes in the push behind that.